notebooks/working/5.0-mab-simulation-causal-graph.md

```diff
@@ -0,0 +1,1432 @@
+                params_dict.setdefault(var, {})['parameters'] = list(BestDict.items())[0][1]
+
+    return params_dict
+```
+
+```python
+observation_id_col = 'observation_id'
+
+alternative_id_col = 'mode_id'
+
+individual_specific_variables = ['household_size','num_kids',
+                                'num_cars','num_licensed_drivers']
+
+alternative_specific_dict = {1:['total_travel_distance'],
+                             2:['total_travel_distance'],
+                             3:['total_travel_distance'],
+                             4:['total_travel_time'],
+                             5:['total_travel_time'],
+                             6:['total_travel_time'],
+                             7:['total_travel_distance'],
+                             8:['total_travel_distance']}
+
+trip_specific_variables = ['cross_bay']
+
+alternative_name_dict = {1: 'drive_alone',
+                         2: 'shared_2',
+                         3: 'shared_3p',
+                         4: 'wtw',
+                         5: 'dtw',
+                         6: 'wtd',
+                         7: 'walk',
+                         8: 'bike'}
+
+variable_type = {'num_kids': 'categorical',
+                 'household_size': 'categorical',
+                 'num_cars': 'categorical',
+                 'num_licensed_drivers': 'categorical',
+                 'cross_bay': 'categorical',
+                 'total_travel_time': 'continuous',
+                 'total_travel_distance': 'continuous',
+                 'total_travel_cost': 'continuous'}
+
+distributions = ['normal', 'alpha', 'beta', 'gamma', 'expon', 'gumbel']
+```
+
+```python
+bike_data_params = DistNodeNoParent(data_long=bike_data_long,
+                                    alt_id_col=alternative_id_col,
+                                    obs_id_col=observation_id_col,
+                                    alt_spec_dic=alternative_specific_dict,
+                                    alt_name_dic=alternative_name_dict,
+                                    ind_spec=individual_specific_variables,
+                                    trip_spec=trip_specific_variables,
+                                    var_types=variable_type,
+                                    cont_dists=distributions)
+```
+
+```python
+bike_data_params
+```
+
+## 3. Regression Parameters function
+
+
+## Define Function to run Regression
+
+```python
+def FitAlternativeRegression(regressions, reg_types, data):
+    
+    regression_results = {}
+    
+    for reg in regressions.keys():
+        if reg_types[reg] == 'linear':
+            data_x = sm.add_constant(data[regressions[reg][0]])
+            data_y = data[regressions[reg][1]]
+
+            model = sm.OLS(data_y, data_x)
+
+            results = model.fit()
+            regression_results[regressions[reg][1]+'_on_'+regressions[reg][0]] = results
+        
+        if reg_types[reg] == 'logistic':
```

Comment:
Right now we can only do logistic regression (i.e. binary features) but we may want multinomial regression for multi-category features.